pkg/apps/store/store.go

```diff
@@ -104,38 +121,62 @@ func GetComponents(v types.PublicVersion, config *protoform.Config, kubeclient *
 	return cp, nil
 }
 
-func generateRc(v types.PublicVersion, config *protoform.Config, kubeclient *kubernetes.Clientset, blackduck *v1.Blackduck) ([]*components.ReplicationController, error) {
-	var rcs []*components.ReplicationController
+func getCR(cr interface{}) (*customResource, error) {
+	namespace, ok := reflect.ValueOf(cr).Elem().FieldByName("Spec").FieldByName("Namespace").Interface().(string)
```

Comment:
You can simplfy by doing 

var, ok := reflect.ValueOf(cr).Elem().FieldByName("Spec)

Then just need to call 

var.FieldByName("Namespace").String() for example rather than the whole chain again.

Reply:
fixed